x/ibc/03-connection/keeper/handshake.go

```diff
@@ -0,0 +1,242 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ics02types "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	ics23 "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+)
+
+// ConnOpenInit initialises a connection attempt on chain A.
+func (k Keeper) ConnOpenInit(
+	ctx sdk.Context,
+	connectionID, // identifier
+	clientID string,
+	counterparty types.Counterparty, // desiredCounterpartyConnectionIdentifier, counterpartyPrefix, counterpartyClientIdentifier
+) error {
+	// TODO: validateConnectionIdentifier(identifier)
```

Comment:
What remains to be done here?

Reply:
This validation is purposely not fixed in the spec, so the SDK can choose what identifiers it wants to allow.